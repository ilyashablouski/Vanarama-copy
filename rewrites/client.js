/* eslint-disable no-console */
const fetch = require('node-fetch');

const {
  ApolloClient,
  createHttpLink,
  InMemoryCache,
} = require('@apollo/client');

// const inspect = require('../inspect');

const httpLink = createHttpLink({
  uri: process.env.API_URL,
  fetch,
  headers: {
    'x-api-key': process.env.API_KEY,
  },
});

const client = new ApolloClient({
  link: httpLink,
  cache: new InMemoryCache(),
  onError: ({ networkError, graphQLErrors }) => {
    console.log('graphQLErrors', graphQLErrors);
    console.log('networkError', networkError);
  },
});

module.exports = client;
