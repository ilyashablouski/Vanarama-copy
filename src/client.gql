extend type Query {
  selectedOrderUuid: ID
  selectedDerivativeId: ID
}
