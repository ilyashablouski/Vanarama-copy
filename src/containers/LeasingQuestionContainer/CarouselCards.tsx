import React, { useState, FC } from 'react';
import dynamic from 'next/dynamic';
import DOMPurify from 'dompurify';
import { GenericPageQuestionQuery_genericPage_sections_cards_cards as ICaruselCard } from '../../../generated/GenericPageQuestionQuery';
import Skeleton from '../../components/Skeleton';

const Button = dynamic(
  () => import('@vanarama/uibook/lib/components/atoms/button'),
  {
    loading: () => <Skeleton count={1} />,
  },
);
const Carousel = dynamic(
  () => import('@vanarama/uibook/lib/components/organisms/carousel'),
  {
    loading: () => <Skeleton count={4} />,
    ssr: false,
  },
);
const Text = dynamic(
  () => import('@vanarama/uibook/lib/components/atoms/text'),
  {
    loading: () => <Skeleton count={1} />,
  },
);
const Card = dynamic(
  () => import('@vanarama/uibook/lib/components/molecules/cards'),
  {
    loading: () => <Skeleton count={5} />,
  },
);

interface ICarouselCards {
  cards: ICaruselCard[];
}

const renderCarouselCards = (cards: (ICaruselCard | null)[]) =>
  cards.map(card => {
    const [readMore, toggleRead] = useState(true);
    const clearHtml = DOMPurify.sanitize(card && card.body ? card.body : '');
    const cardText = {
      dangerouslySetInnerHTML: {
        __html: readMore ? `${clearHtml.slice(0, 120)}...` : clearHtml,
      },
    };

    return card?.title && card.body && card.name ? (
      <Card
        optimisedHost={process.env.IMG_OPTIMISATION_HOST}
        key={card.name}
        title={{ title: card?.title }}
        imageSrc={card.image?.file?.url}
      >
        <Text size="regular" color="dark">
          <span {...cardText} />
        </Text>
        <Button
          fill="clear"
          color="teal"
          size="regular"
          label={readMore ? 'Read More' : 'Read Less'}
          onClick={() => toggleRead(!readMore)}
        />
      </Card>
    ) : null;
  });

const CarouselCards: FC<ICarouselCards> = ({ cards }) => (
  <Carousel className="-col3" countItems={3}>
    {renderCarouselCards(cards)}
  </Carousel>
);

export default CarouselCards;
