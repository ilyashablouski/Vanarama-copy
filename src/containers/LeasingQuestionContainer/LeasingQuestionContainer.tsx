import React, { FC } from 'react';
import Heading from '@vanarama/uibook/lib/components/atoms/heading';
import Text from '@vanarama/uibook/lib/components/atoms/text';
import Card from '@vanarama/uibook/lib/components/molecules/cards';
import BreadCrumb from '@vanarama/uibook/lib/components/atoms/breadcrumb';
import Button from '@vanarama/uibook/lib/components/atoms/button';
import Router from 'next/router';
import ReactMarkdown from 'react-markdown';
import Carousel from '@vanarama/uibook/lib/components/organisms/carousel';
import {
  GenericPageQuestionQuery_genericPage_sections_cards_cards as ICaruselCard,
  GenericPageQuestionQuery_genericPage_sections as Section,
  GenericPageQuestionQuery_genericPage_sections_faqs_questionSets_questionAnswers as IQuestion,
} from '../../../generated/GenericPageQuestionQuery';
import getTitleTag from '../../utils/getTitleTag';
import ArrowForwardSharp from '@vanarama/uibook/lib/assets/icons/ArrowForwardSharp';
import Accordion from '@vanarama/uibook/lib/components/molecules/accordion/Accordion';
import RouterLink from 'components/RouterLink/RouterLink';

interface IProps {
  sections: Section | null;
  title: string | null;
  body: string | null;
  crumbs: { href: string; label: string }[];
}

const renderCarouselCards = (cards: (ICaruselCard | null)[]) =>
  cards.map(card =>
    card?.title && card.body && card.name ? (
      <Card
        key={card.name}
        className="card__article"
        title={{ title: card?.title }}
        imageSrc={card.image?.file?.url}
      >
        <Text size="regular" color="dark">
          {card.body || ''}
        </Text>
        <Button
          fill="clear"
          color="teal"
          size="regular"
          label={card?.link?.text}
          onClick={() => {
            Router.push(card?.link?.url || '');
          }}
        />
      </Card>
    ) : null,
  );

const accordionItems = (questions: (IQuestion | null)[] | undefined | null) => {
  debugger;
  return questions ? questions.map((item: any) => ({
    id: item.question,
    key: item.question,
    title: item.question,
    children: (
      <ReactMarkdown
        source={item.answer || ''}
        renderers={{
          link: props => {
            const { href, children } = props;
            return <RouterLink link={{ href, label: children }} />;
          },
        }}
      />)
  })) : [];
};

const renderQuestions = (questions: (IQuestion | null)[] | undefined | null) => {
  questions?.length ? (
    <Accordion items={accordionItems(questions)} />
  ) : (
      <p>Error: No questions</p>
    )
}


const LeasingQuestionContainer: FC<IProps> = ({
  body,
  title,
  sections,
  crumbs,
}) => {
  const carousel = sections?.cards;
  const questionSet = sections?.questionSet;
  return (
    <>
      <div className="row:title">
        <BreadCrumb items={crumbs} />
        <Heading tag="h1" size="xlarge" color="black">
          {title}
        </Heading>
        <ReactMarkdown source={body || ''} />
        <Button
          type="button"
          label={'Ask A Question'}
          disabled={false}
          color="primary"
          icon={<ArrowForwardSharp />}
          iconColor="white"
          iconPosition="after"
          dataTestId="ask-a-question"
        />
      </div>
      <div className="row:bg-lighter">
        <div className="row:carousel">
          <Heading tag={getTitleTag(carousel?.titleTag || null) as keyof JSX.IntrinsicElements} size="large" color="black">
            {carousel?.title || ''}
          </Heading>
          {carousel?.cards && (
            <Carousel className="-col3" countItems={3}>
              {renderCarouselCards(carousel?.cards)}
            </Carousel>
          )}
        </div>
      </div>
      <div className="row:bg-lighter">
        <div className="row:cards-3col">
          <Heading tag="h2" size="large" color="black">
            {questionSet?.title || ''}
          </Heading>
          {renderQuestions(questionSet?.questionAnswers)}
        </div>
      </div>
    </>
  );
};

export default LeasingQuestionContainer;
