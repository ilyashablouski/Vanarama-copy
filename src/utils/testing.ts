// eslint-disable-next-line import/no-extraneous-dependencies
import { fireEvent } from '@testing-library/react';

// eslint-disable-next-line import/prefer-default-export
export const inputChange = (element: Element, value: string) => {
  fireEvent.input(element, {
    target: { value },
  });
};
