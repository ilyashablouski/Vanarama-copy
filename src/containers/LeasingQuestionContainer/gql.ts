import { gql, useQuery } from '@apollo/client';
import {
  GenericPageQuestionQuery,
  GenericPageQuestionQueryVariables,
} from '../../../generated/GenericPageQuestionQuery';

export const GENERIC_PAGE_QUESTION = gql`
  query GenericPageQuestionQuery($slug: String!) {
    genericPage(slug: $slug) {
      id
      metaData {
        title
        name
        metaRobots
        metaDescription
        legacyUrl
        pageType
        canonicalUrl
        slug
        schema
        publishedOn
        breadcrumbs
      }
      sections {
        hero {
          title
          body
          image {
            file {
              url
            }
          }
          heroCard {
            title
            body
          }
        }
        tiles {
          position
          name
          tilesTitle
          titleTag
        }
        leadText {
          titleTag
          heading
          description
          position
        }
        iconBullets1 {
          title
          iconBullets {
            text
          }
        }
        iconBullets2 {
          title
          iconBullets {
            text
          }
        }
        faqs {
          title
          body
          questionSets {
            title
            questionAnswers {
              question
              answer
            }
          }
        }
        cards {
          position
          name
          titleTag
          description
          title
          cards {
            title
            name
            image {
              title
              description
              file {
                url
                fileName
              }
            }
            body
            titleTag
            link {
              text
              url
              legacyUrl
            }
          }
        }
        hero {
          position
          flag
          title
          titleTag
          body
          image {
            title
            description
            file {
              url
              fileName
              contentType
            }
          }
          heroCard {
            title
            body
          }
        }
        leadText {
          position
          heading
          titleTag
          description
        }
        rowText {
          position
          heading
          titleTag
          subHeading
          body
        }
        featured {
          layout
          body
          title
          video
          link {
            text
            url
            legacyUrl
          }
          image {
            title
            description
            file {
              url
              fileName
              contentType
              details {
                size
                image {
                  width
                  height
                }
              }
            }
          }
        }
        featured1 {
          layout
          body
          titleTag
          title
          link {
            url
            text
            legacyUrl
          }
          image {
            title
            description
            file {
              url
              fileName
            }
          }
        }
        featured2 {
          layout
          body
          titleTag
          link {
            url
            text
            legacyUrl
          }
          image {
            title
            description
            file {
              url
              fileName
            }
          }
          title
          cards {
            name
            title
            image {
              title
              description
              file {
                url
                fileName
              }
            }
            body
            link {
              text
              url
              legacyUrl
            }
          }
        }
        carousel {
          title
          name
          cards {
            name
            title
            image {
              title
              description
              file {
                url
                fileName
              }
            }
            body
            link {
              text
              url
              legacyUrl
            }
          }
        }
        tiles {
          position
          name
          tilesTitle
          titleTag
          tiles {
            body
            title
            link {
              text
              url
              legacyUrl
            }
            image {
              title
              description
              file {
                url
                fileName
              }
            }
          }
        }
        questionSet {
          title
          questionAnswers {
            answer
            question
          }
        }
      }
      intro
      body
      featuredImage {
        title
        description
        file {
          url
          fileName
          contentType
        }
      }
    }
  }
`;

export function useGenericPageQuestion(slug: string) {
  return useQuery<GenericPageQuestionQuery, GenericPageQuestionQueryVariables>(
    GENERIC_PAGE_QUESTION,
    {
      variables: {
        slug,
      },
    },
  );
}
