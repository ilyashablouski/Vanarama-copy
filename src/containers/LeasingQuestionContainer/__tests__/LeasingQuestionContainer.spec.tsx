import React from 'react';
import preloadAll from 'jest-next-dynamic';
import { MockedProvider } from '@apollo/client/testing';
import { screen, render, waitFor } from '@testing-library/react';
import LeasingQuestionContainer from '../LeasingQuestionContainer';

jest.mock('next/router', () => ({
  useRouter: () => ({
    asPath: '/',
  }),
}));

// ARRANG
const METADATA = {
  title: 'Insurance, Tax & Vat Leasing Questions',
  metaRobots: null,
  metaDescription: 'Insurance Tax Vat',
  publishedOn: null,
  legacyUrl:
    'https://www.vanarama.com/van-leasing-questions/insurance-tax-vat.html',
  pageType: 'Customer Leasing Questions',
  __typename: 'Meta',
};
const BODY = ``;

const SECTIONS = {
  questionSet: {
    name: 'All Questions',
    questionAnswers: [
      {
        question:
          "I would like a large van, but still unsure what is the best purchase option? Also what is the best tax relief option through my limited company? I'm not VAT registered, so can I claim back my contract payments?",
        answer:
          'Our expert Mike answered... Hello Adrian, thank you very much for leaving your question on our website. There are various leasing options available depending on what you would like to achieve at the end of the contract. If you are not VAT registered then you will not be able to claim the VAT back on the rentals, however depending on which contract you decided to go for, the rentals may well be 100% tax deductible. If you would like to discuss any of these options in greater detail, the best thing to do is contact one of our specialists on 01442 838193 or alternatively please let us know the best contact number for you and when would be a convenient time to contact you. Many thanks once again for leaving your question. Kind regards, Mike Daly.',
      },
      {
        question:
          "I would like a large van, but still unsure what is the best purchase option? Also what is the best tax relief option through my limited company? I'm not VAT registered, so can I claim back my contract payments?",
        answer:
          'Our expert Mike answered... Hello Adrian, thank you very much for leaving your question on our website. There are various leasing options available depending on what you would like to achieve at the end of the contract. If you are not VAT registered then you will not be able to claim the VAT back on the rentals, however depending on which contract you decided to go for, the rentals may well be 100% tax deductible. If you would like to discuss any of these options in greater detail, the best thing to do is contact one of our specialists on 01442 838193 or alternatively please let us know the best contact number for you and when would be a convenient time to contact you. Many thanks once again for leaving your question. Kind regards, Mike Daly.',
      },
      {
        question:
          "I would like a large van, but still unsure what is the best purchase option? Also what is the best tax relief option through my limited company? I'm not VAT registered, so can I claim back my contract payments?",
        answer:
          'Our expert Mike answered... Hello Adrian, thank you very much for leaving your question on our website. There are various leasing options available depending on what you would like to achieve at the end of the contract. If you are not VAT registered then you will not be able to claim the VAT back on the rentals, however depending on which contract you decided to go for, the rentals may well be 100% tax deductible. If you would like to discuss any of these options in greater detail, the best thing to do is contact one of our specialists on 01442 838193 or alternatively please let us know the best contact number for you and when would be a convenient time to contact you. Many thanks once again for leaving your question. Kind regards, Mike Daly.',
      },
      {
        question:
          "I would like a large van, but still unsure what is the best purchase option? Also what is the best tax relief option through my limited company? I'm not VAT registered, so can I claim back my contract payments?",
        answer:
          'Our expert Mike answered... Hello Adrian, thank you very much for leaving your question on our website. There are various leasing options available depending on what you would like to achieve at the end of the contract. If you are not VAT registered then you will not be able to claim the VAT back on the rentals, however depending on which contract you decided to go for, the rentals may well be 100% tax deductible. If you would like to discuss any of these options in greater detail, the best thing to do is contact one of our specialists on 01442 838193 or alternatively please let us know the best contact number for you and when would be a convenient time to contact you. Many thanks once again for leaving your question. Kind regards, Mike Daly.',
      },
    ],
  },
  cards: {
    position: 2,
    name: 'Most Popular Questions',
    titleTag: null,
    description: null,
    cards: [
      {
        body:
          'Our advertised prices will not include servicing and it is your responsibility to make sure that the' +
          " vehicle is serviced in line with the vehicle's servicing schedule. However, we are able to offer you a optional package called Service Plus which covers all routine services, mechanical repairs, replacement tires, MOT if required, replacement brakes, wiper blades and bulbs (all subject to fair wear and tear). The additional cost of the Service Plus package depends on the annual mileage you select and how long you decide to lease the vehicle for.",
        image: null,
        link: null,
        name: 'Question 1',
        title: 'Am I Responsible For The Servicing Of My Leased Van?',
        titleTag: 'h3',
      },
      {
        body:
          'Depending on the amount of damage done to the vehicle, the best thing to do is speak with your' +
          ' insurance company directly and arrange for the repair to be carried out by an approved body shop. If the damage is more of a scrape or scuff, then you could contact a mobile smart repairer who can come to you and repair the vehicle on-site or at your home address.',
        image: null,
        link: null,
        name: 'Question 2',
        title: 'I Have Damaged My Van, What Do I Need To Do Next?',
        titleTag: 'h3',
      },
      {
        body:
          'You can have your van serviced at any independent garage providing they are VAT registered, qualified' +
          " to work on your vehicle & use genuine manufacturer parts for the service. Doing this will keep the vehicle's warranty valid until its standard expiry date.",
        image: null,
        link: null,
        name: 'Question 3',
        title: 'What Type Of Garage Should I Use For Servicing My Van?',
        titleTag: 'h3',
      },
    ],
  },
} as any;

describe('<LeasingQuestionContainer />', () => {
  beforeAll(async () => {
    await preloadAll();
  });
  it('should match snapshot', async () => {
    // ACT
    const getComponent = render(
      <MockedProvider addTypename={false}>
        <LeasingQuestionContainer
          sections={SECTIONS}
          title={METADATA.title}
          body={BODY}
        />
      </MockedProvider>,
    );
    // ASSERT
    await waitFor(() => {
      expect(
        screen.getAllByText(
          `Am I Responsible For The Servicing Of My Leased Van?`,
        ),
      );
    });

    const tree = getComponent.baseElement;
    expect(tree).toMatchSnapshot();
  });
});
