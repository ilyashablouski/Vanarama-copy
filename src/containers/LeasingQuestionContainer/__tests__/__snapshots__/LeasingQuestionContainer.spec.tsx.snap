// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeasingQuestionContainer /> should match snapshot 1`] = `
<body>
  <div>
    <div
      class="row:title"
    >
      <h1
        class="heading -black -xlarge"
      >
        Insurance, Tax & Vat Leasing Questions
      </h1>
    </div>
    <div
      class="row:bg-lighter"
    >
      <div
        class="row:carousel"
      >
        <p
          class="heading -black -large"
        >
          Most Popular Questions
        </p>
        <div
          class="carousel -col3"
        >
          <div
            class="swiper-container swiper-container-initialized swiper-container-horizontal"
          >
            <div
              class="swiper-wrapper"
              style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
            >
              <div
                class="swiper-slide swiper-slide-duplicate"
                data-swiper-slide-index="0"
              >
                <div
                  class="card"
                  data-testid="card"
                >
                  <div
                    class="title"
                  >
                    <span
                      class="heading -black -lead"
                      data-testid="card-heading"
                    >
                      Am I Responsible For The Servicing Of My Leased Van?
                    </span>
                  </div>
                  <span
                    class="text -regular -dark"
                  >
                    <span>
                      Our advertised prices will not include servicing and it is your responsibility to make sure that the vehicle is serviced...
                    </span>
                  </span>
                  <button
                    class="button -teal -regular -clear"
                  >
                    <div
                      class="button--inner"
                    >
                      Read More
                    </div>
                  </button>
                </div>
              </div>
              <div
                class="swiper-slide swiper-slide-duplicate"
                data-swiper-slide-index="1"
              >
                <div
                  class="card"
                  data-testid="card"
                >
                  <div
                    class="title"
                  >
                    <span
                      class="heading -black -lead"
                      data-testid="card-heading"
                    >
                      I Have Damaged My Van, What Do I Need To Do Next?
                    </span>
                  </div>
                  <span
                    class="text -regular -dark"
                  >
                    <span>
                      Depending on the amount of damage done to the vehicle, the best thing to do is speak with your insurance company directl...
                    </span>
                  </span>
                  <button
                    class="button -teal -regular -clear"
                  >
                    <div
                      class="button--inner"
                    >
                      Read More
                    </div>
                  </button>
                </div>
              </div>
              <div
                class="swiper-slide swiper-slide-duplicate"
                data-swiper-slide-index="2"
              >
                <div
                  class="card"
                  data-testid="card"
                >
                  <div
                    class="title"
                  >
                    <span
                      class="heading -black -lead"
                      data-testid="card-heading"
                    >
                      What Type Of Garage Should I Use For Servicing My Van?
                    </span>
                  </div>
                  <span
                    class="text -regular -dark"
                  >
                    <span>
                      You can have your van serviced at any independent garage providing they are VAT registered, qualified to work on your ve...
                    </span>
                  </span>
                  <button
                    class="button -teal -regular -clear"
                  >
                    <div
                      class="button--inner"
                    >
                      Read More
                    </div>
                  </button>
                </div>
              </div>
              <div
                class="swiper-slide"
                data-swiper-slide-index="0"
              >
                <div
                  class="card"
                  data-testid="card"
                >
                  <div
                    class="title"
                  >
                    <span
                      class="heading -black -lead"
                      data-testid="card-heading"
                    >
                      Am I Responsible For The Servicing Of My Leased Van?
                    </span>
                  </div>
                  <span
                    class="text -regular -dark"
                  >
                    <span>
                      Our advertised prices will not include servicing and it is your responsibility to make sure that the vehicle is serviced...
                    </span>
                  </span>
                  <button
                    class="button -teal -regular -clear"
                  >
                    <div
                      class="button--inner"
                    >
                      Read More
                    </div>
                  </button>
                </div>
              </div>
              <div
                class="swiper-slide"
                data-swiper-slide-index="1"
              >
                <div
                  class="card"
                  data-testid="card"
                >
                  <div
                    class="title"
                  >
                    <span
                      class="heading -black -lead"
                      data-testid="card-heading"
                    >
                      I Have Damaged My Van, What Do I Need To Do Next?
                    </span>
                  </div>
                  <span
                    class="text -regular -dark"
                  >
                    <span>
                      Depending on the amount of damage done to the vehicle, the best thing to do is speak with your insurance company directl...
                    </span>
                  </span>
                  <button
                    class="button -teal -regular -clear"
                  >
                    <div
                      class="button--inner"
                    >
                      Read More
                    </div>
                  </button>
                </div>
              </div>
              <div
                class="swiper-slide"
                data-swiper-slide-index="2"
              >
                <div
                  class="card"
                  data-testid="card"
                >
                  <div
                    class="title"
                  >
                    <span
                      class="heading -black -lead"
                      data-testid="card-heading"
                    >
                      What Type Of Garage Should I Use For Servicing My Van?
                    </span>
                  </div>
                  <span
                    class="text -regular -dark"
                  >
                    <span>
                      You can have your van serviced at any independent garage providing they are VAT registered, qualified to work on your ve...
                    </span>
                  </span>
                  <button
                    class="button -teal -regular -clear"
                  >
                    <div
                      class="button--inner"
                    >
                      Read More
                    </div>
                  </button>
                </div>
              </div>
              <div
                class="swiper-slide swiper-slide-duplicate"
                data-swiper-slide-index="0"
              >
                <div
                  class="card"
                  data-testid="card"
                >
                  <div
                    class="title"
                  >
                    <span
                      class="heading -black -lead"
                      data-testid="card-heading"
                    >
                      Am I Responsible For The Servicing Of My Leased Van?
                    </span>
                  </div>
                  <span
                    class="text -regular -dark"
                  >
                    <span>
                      Our advertised prices will not include servicing and it is your responsibility to make sure that the vehicle is serviced...
                    </span>
                  </span>
                  <button
                    class="button -teal -regular -clear"
                  >
                    <div
                      class="button--inner"
                    >
                      Read More
                    </div>
                  </button>
                </div>
              </div>
              <div
                class="swiper-slide swiper-slide-duplicate"
                data-swiper-slide-index="1"
              >
                <div
                  class="card"
                  data-testid="card"
                >
                  <div
                    class="title"
                  >
                    <span
                      class="heading -black -lead"
                      data-testid="card-heading"
                    >
                      I Have Damaged My Van, What Do I Need To Do Next?
                    </span>
                  </div>
                  <span
                    class="text -regular -dark"
                  >
                    <span>
                      Depending on the amount of damage done to the vehicle, the best thing to do is speak with your insurance company directl...
                    </span>
                  </span>
                  <button
                    class="button -teal -regular -clear"
                  >
                    <div
                      class="button--inner"
                    >
                      Read More
                    </div>
                  </button>
                </div>
              </div>
              <div
                class="swiper-slide swiper-slide-duplicate"
                data-swiper-slide-index="2"
              >
                <div
                  class="card"
                  data-testid="card"
                >
                  <div
                    class="title"
                  >
                    <span
                      class="heading -black -lead"
                      data-testid="card-heading"
                    >
                      What Type Of Garage Should I Use For Servicing My Van?
                    </span>
                  </div>
                  <span
                    class="text -regular -dark"
                  >
                    <span>
                      You can have your van serviced at any independent garage providing they are VAT registered, qualified to work on your ve...
                    </span>
                  </span>
                  <button
                    class="button -teal -regular -clear"
                  >
                    <div
                      class="button--inner"
                    >
                      Read More
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="carousel--nav"
            >
              <button
                class="carousel--nav-arrow swiper-prev"
                type="button"
              >
                <i
                  class="icon -darker -regular"
                >
                  <svg
                    height="1em"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <polyline
                      points="244 400 100 256 244 112"
                      style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                    />
                    <line
                      style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                      x1="120"
                      x2="412"
                      y1="256"
                      y2="256"
                    />
                  </svg>
                </i>
              </button>
              <div
                class="swiper-pagination swiper-pagination-clickable swiper-pagination-bullets"
              />
              <button
                class="carousel--nav-arrow swiper-next"
                type="button"
              >
                <i
                  class="icon -darker -regular"
                >
                  <svg
                    height="1em"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <polyline
                      points="268 112 412 256 268 400"
                      style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                    />
                    <line
                      style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                      x1="392"
                      x2="100"
                      y1="256"
                      y2="256"
                    />
                  </svg>
                </i>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row:bg-lighter"
    >
      <div
        class="row:lead-text"
      >
        <h2
          class="heading -black -large"
        />
        <ul
          class="tilebox accordion"
        >
          <li
            class="accordion-item"
            data-testid="accordion-item"
          >
            <button
              class="accordion--heading"
              type="button"
            >
              <i
                class="icon accordion-item--chevron"
              >
                <svg
                  height="1em"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <polyline
                    points="184 112 328 256 184 400"
                    style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                  />
                </svg>
              </i>
              <div
                class="accordion--title"
              >
                <span
                  class="text -regular -black"
                >
                  I would like a large van, but still unsure what is the best purchase option? Also what is the best tax relief option through my limited company? I'm not VAT registered, so can I claim back my contract payments?
                </span>
              </div>
            </button>
            <div
              class="accordion--content"
            >
              <p>
                Our expert Mike answered... Hello Adrian, thank you very much for leaving your question on our website. There are various leasing options available depending on what you would like to achieve at the end of the contract. If you are not VAT registered then you will not be able to claim the VAT back on the rentals, however depending on which contract you decided to go for, the rentals may well be 100% tax deductible. If you would like to discuss any of these options in greater detail, the best thing to do is contact one of our specialists on 01442 838193 or alternatively please let us know the best contact number for you and when would be a convenient time to contact you. Many thanks once again for leaving your question. Kind regards, Mike Daly.
              </p>
            </div>
          </li>
          <li
            class="accordion-item"
            data-testid="accordion-item"
          >
            <button
              class="accordion--heading"
              type="button"
            >
              <i
                class="icon accordion-item--chevron"
              >
                <svg
                  height="1em"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <polyline
                    points="184 112 328 256 184 400"
                    style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                  />
                </svg>
              </i>
              <div
                class="accordion--title"
              >
                <span
                  class="text -regular -black"
                >
                  I would like a large van, but still unsure what is the best purchase option? Also what is the best tax relief option through my limited company? I'm not VAT registered, so can I claim back my contract payments?
                </span>
              </div>
            </button>
            <div
              class="accordion--content"
            >
              <p>
                Our expert Mike answered... Hello Adrian, thank you very much for leaving your question on our website. There are various leasing options available depending on what you would like to achieve at the end of the contract. If you are not VAT registered then you will not be able to claim the VAT back on the rentals, however depending on which contract you decided to go for, the rentals may well be 100% tax deductible. If you would like to discuss any of these options in greater detail, the best thing to do is contact one of our specialists on 01442 838193 or alternatively please let us know the best contact number for you and when would be a convenient time to contact you. Many thanks once again for leaving your question. Kind regards, Mike Daly.
              </p>
            </div>
          </li>
          <li
            class="accordion-item"
            data-testid="accordion-item"
          >
            <button
              class="accordion--heading"
              type="button"
            >
              <i
                class="icon accordion-item--chevron"
              >
                <svg
                  height="1em"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <polyline
                    points="184 112 328 256 184 400"
                    style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                  />
                </svg>
              </i>
              <div
                class="accordion--title"
              >
                <span
                  class="text -regular -black"
                >
                  I would like a large van, but still unsure what is the best purchase option? Also what is the best tax relief option through my limited company? I'm not VAT registered, so can I claim back my contract payments?
                </span>
              </div>
            </button>
            <div
              class="accordion--content"
            >
              <p>
                Our expert Mike answered... Hello Adrian, thank you very much for leaving your question on our website. There are various leasing options available depending on what you would like to achieve at the end of the contract. If you are not VAT registered then you will not be able to claim the VAT back on the rentals, however depending on which contract you decided to go for, the rentals may well be 100% tax deductible. If you would like to discuss any of these options in greater detail, the best thing to do is contact one of our specialists on 01442 838193 or alternatively please let us know the best contact number for you and when would be a convenient time to contact you. Many thanks once again for leaving your question. Kind regards, Mike Daly.
              </p>
            </div>
          </li>
          <li
            class="accordion-item"
            data-testid="accordion-item"
          >
            <button
              class="accordion--heading"
              type="button"
            >
              <i
                class="icon accordion-item--chevron"
              >
                <svg
                  height="1em"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <polyline
                    points="184 112 328 256 184 400"
                    style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                  />
                </svg>
              </i>
              <div
                class="accordion--title"
              >
                <span
                  class="text -regular -black"
                >
                  I would like a large van, but still unsure what is the best purchase option? Also what is the best tax relief option through my limited company? I'm not VAT registered, so can I claim back my contract payments?
                </span>
              </div>
            </button>
            <div
              class="accordion--content"
            >
              <p>
                Our expert Mike answered... Hello Adrian, thank you very much for leaving your question on our website. There are various leasing options available depending on what you would like to achieve at the end of the contract. If you are not VAT registered then you will not be able to claim the VAT back on the rentals, however depending on which contract you decided to go for, the rentals may well be 100% tax deductible. If you would like to discuss any of these options in greater detail, the best thing to do is contact one of our specialists on 01442 838193 or alternatively please let us know the best contact number for you and when would be a convenient time to contact you. Many thanks once again for leaving your question. Kind regards, Mike Daly.
              </p>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;
